pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -364,11 +371,13 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 	if len(instanceID) == 0 {
 		return reconcile.Result{}, errors.Errorf("unable to get instance ID from provider ID for machine %s", machine.Name)
 	}
-	providerName := strings.TrimSuffix(providerTokens[0], ":")
 
 	log.Info("processing", "namespace", request.Namespace, "name", request.Name)
 	// Make the Machine a Windows Worker node
-	if err := r.addWorkerNode(ipAddress, providerName, instanceID); err != nil {
+	if len(machine.Name) == 0 {
+		return reconcile.Result{}, nil
+	}
```

Comment:
Is this possible? @aravindhp

Reply:
It is unlikely. You cannot create any k8s object without a [Name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) AFAIK. Not sure if @ravisantoshgudimetla was just doing defensive programming here or what his intention was.